libraries/chain/protocol/transaction.cpp

```diff
@@ -307,20 +305,17 @@ const flat_set<public_key_type>& signed_transaction::get_signature_keys( const c
 { try {
    // Strictly we should check whether the given chain ID is same as the one used to initialize the `signees` field.
    // However, we don't pass in another chain ID so far, for better performance, we skip the check.
-   if( signees.empty() && !signatures.empty() )
+   auto d = sig_digest( chain_id );
+   flat_set<public_key_type> result;
```

Comment:
That wouldn't help, because then the set has to be re-sorted when assigning to `_signees` field.
Unless we also change the sorting order there, which means the signature of this method has to be changed as well, and all places that use it... it simply doesn't pay out, see #1401 .